src/transformers/quantizers/quantizer_compressed_tensors.py

```diff
@@ -124,6 +124,21 @@ def _process_model_before_weight_loading(self, model, **kwargs):
         elif not self.quantization_config.is_quantization_compressed:
             apply_quantization_config(model, ct_quantization_config)
 
+        # Identify quantized modules with integer weights/activations
+        quant_targets = set()
+        if ct_quantization_config:
+            for group in ct_quantization_config.config_groups.values():
+                if group.weights.type.startswith("int") or (
+                    group.input_activations and group.input_activations.type.startswith("int")
+                ):
+                    quant_targets.update(group.targets)
+
+        # Disable gradient computation for quantized int modules
+        for module in model.modules():
```

Comment:
just an fyi - we only want the weights to have their weight gradient disabled until the model is loaded.